File: Documentation/3.0/Samples/csharp/Raven.Documentation.Samples/ClientApi/Session/HowTo/MarkAsReadonly.cs
using Raven.Client.Document;
using Raven.Documentation.CodeSamples.Orders;

namespace Raven.Documentation.Samples.ClientApi.Session.HowTo
{
	public class MarkAsReadonly
	{
		private interface IFoo
		{
			#region mark_as_readonly_1
			void MarkReadOnly(object entity);
			#endregion
		}

		public MarkAsReadonly()
		{
			using (var store = new DocumentStore())
			{
				using (var session = store.OpenSession())
				{
					#region mark_as_readonly_2
					var employee = session.Load<Employee>("employees/1");
					session.Advanced.MarkReadOnly(employee);
					session.SaveChanges();
					#endregion
				}
			}
		}
	}
}

File: Documentation/3.5/Samples/java/src/test/java/net/ravendb/clientapi/session/howto/MarkAsReadonly.java
package net.ravendb.clientapi.session.howto;

import net.ravendb.client.IDocumentSession;
import net.ravendb.client.IDocumentStore;
import net.ravendb.client.document.DocumentStore;
import net.ravendb.samples.northwind.Employee;


public class MarkAsReadonly {
  @SuppressWarnings("unused")
  private interface IFoo {
    //region mark_as_readonly_1
    public void markReadOnly(Object entity);
    //endregion
  }

  public MarkAsReadonly() throws Exception {
    try (IDocumentStore store = new DocumentStore()) {
      try (IDocumentSession session = store.openSession()) {
        //region mark_as_readonly_2
        Employee employee = session.load(Employee.class, "employees/1");
        session.advanced().markReadOnly(employee);
        session.saveChanges();
        //endregion
      }
    }
  }
}
